// @Helpers/responses.js
module.exports = {
  successResponseData(callback, data, code = 1, message, extras, statusCode=200) {
    const response = {
      data,
      meta: {
        code,
        message,
      },
    };
    if (extras) {
      Object.keys(extras).forEach((key) => {
        if ({}.hasOwnProperty.call(extras, key)) {
          response.meta[key] = extras[key];
        }
      });
    }

    return callback(null, {
        body: JSON.stringify(response),
        statusCode,
        headers: {
          "Access-Control-Allow-Origin": "*",
          "Access-Control-Allow-Methods": "*",
          "Access-Control-Allow-Headers": "*",
        },
    });
  },

  successResponseWithoutData(callback, message, code = 1, statusCode=200) {
    const response = {
      data: null,
      meta: {
        code,
        message,
      },
    };
    return callback(null, {
        body: JSON.stringify(response),
      statusCode,
      headers: {
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "*",
        "Access-Control-Allow-Headers": "*",
      },
    });
  },

  errorResponseWithoutData(
    callback,
    message,
    code = 0,
    statusCode = 500,
    metaData = {}
  ) {
    const response = {
      data: null,
      meta: {
        code,
        message,
        ...metaData,
      },
    };
    return callback(null, {
        body: JSON.stringify(response),
      statusCode,
      headers: {
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "*",
        "Access-Control-Allow-Headers": "*",
      },
    });
  },


  validationErrorResponseData(callback, message, code = 400) {
    const response = {
      code,
      message,
    };
    return callback(null, {
        body: JSON.stringify(response),
      statusCode: code,
      headers: {
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "*",
        "Access-Control-Allow-Headers": "*",
      },
    });
  },
};
