// @AwsHelpers/SSM/index.js
const AWS = require( 'aws-sdk' )

const ssm = new AWS.SSM( {
	apiVersion: '2014-11-06',
	region: process.env.Region && process.env.Region !== 'undefined' ? process.env.Region : 'eu-north-1'
} )

async function getByName ( name ) {
	try {
		console.log( 'trying to getByName - name = ' + name )
		var params = {
			Name: `/${process.env.Application}/${process.env.Stage}/${name}`,
			WithDecryption: true
		}
		var request = await ssm.getParameter( params ).promise()
		return request.Parameter.Value
	} catch ( error ) {
		console.error( 'Failed to retrieve param' )
		console.error( error )
		throw error
	}
}
async function getByPath ( path ) {
	try {
		let fullPath = `/${process.env.Application}/${process.env.Stage}/${path}`
		var params = {
			Path: fullPath,
			WithDecryption: true
		}
		var request = await ssm.getParametersByPath( params ).promise()
		let returnParams = request.Parameters.reduce( function ( result, item ) {
			result[item.Name.substring( fullPath.length + 1 )] = item.Value
			return result
		}, {} )
		return returnParams
	} catch ( error ) {
		console.error( 'Failed to retrieve params' )
		console.error( error )
		throw error
	}
}

module.exports = {
	getByName,
	getByPath
}
